### PR TITLE
Included prometheus server for guacql

### PR DESCRIPTION
**Pull request by naveensrinivasan**
# Description of the PR

- Included Prometheus for `guacql`
- Will include in all other https://github.com/guacsec/guac/pull/1500#issuecomment-1823130996 if this is acceptable.

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/cli/init.go [hunk 1]
```diff
@@ -67,3 +70,40 @@ func InitConfig() {
 		logger.Infof("Using config file: %s", viper.ConfigFileUsed())
 	}
 }
+
+// SetupPrometheus sets up the prometheus server
+func SetupPrometheus(ctx context.Context, logger *zap.SugaredLogger, name string) error {
```
Comment (pxp928) [inline pkg/cli/init.go:75]: nit: can grab the logger from the `ctx`, `logger := logging.FromContext(ctx)`

**Review by pxp928 — APPROVED** [review]
LGTM! small nit

File: pkg/cli/init.go (hunk 1 above)
Comment (naveensrinivasan) [reply pkg/cli/init.go:75]: It is being passed to the `func` from the server `logger := logging.FromContext(ctx)` https://github.com/naveensrinivasan/guac/blob/3e8614d14bb9585600e5ee1d6afc9f38ad6b7d57/cmd/guacgql/cmd/server.go#L70

Let me know

**Comment by m-brophy** [comment]
Hey @naveensrinivasan the project I'm on is interested in metrics/timings of guacql method calls. Am I right in thinking this PR sets up the framework for that but doesn't provide the timing info I'm talking about? I ask because if this is so I'll do it myself, I'm just kinda checking I understand correctly.

**Review by lumjjb — APPROVED** [review]

**Review by pxp928 — COMMENTED** [review]
Just need a rebase and we can merge.

**Comment by pxp928** [comment]
> Hey @naveensrinivasan the project I'm on is interested in metrics/timings of guacql method calls. Am I right in thinking this PR sets up the framework for that but doesn't provide the timing info I'm talking about? I ask because if this is so I'll do it myself, I'm just kinda checking I understand correctly.

@m-brophy This should handle this. Correct me if I am wrong @naveensrinivasan.

**Comment by naveensrinivasan** [comment]
> Just need a rebase and we can merge.

👍

**Comment by naveensrinivasan** [comment]
https://github.com/guacsec/guac/pull/1675

**Comment by m-brophy** [comment]
Thanks @naveensrinivasan

**Pull request merged**